specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2021-07-01-preview/virtualNetworks.json

```diff
@@ -0,0 +1,575 @@
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Lists all virtualnetworks under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/virtualnetworksListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "virtualnetworksProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "properties": {
+        "networkType": {
+          "type": "string",
+          "description": "Type of the network"
+        },
+        "subnets": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "Name - The name of the resource that is unique within a resource group. This name can be used to access the resource."
+              },
+              "properties": {
+                "type": "object",
+                "x-ms-client-flatten": true,
+                "properties": {
+                  "addressPrefix": {
+                    "type": "string",
+                    "description": "Cidr for this subnet - IPv4, IPv6"
+                  },
+                  "addressPrefixes": {
+                    "type": "array",
+                    "items": {
+                      "type": "string"
+                    },
+                    "description": "AddressPrefixes - List of address prefixes for the subnet."
+                  },
+                  "ipAllocationMethod": {
+                    "type": "string",
+                    "description": "IPAllocationMethod - The IP address allocation method. Possible values include: 'Static', 'Dynamic'"
+                  },
+                  "ipConfigurationReferences": {
+                    "type": "array",
+                    "items": {
+                      "type": "object",
+                      "properties": {
+                        "ID": {
+                          "type": "string",
+                          "description": "IPConfigurationID"
+                        }
+                      },
+                      "description": "IPConfigurationReference - Describes a IPConfiguration under the virtual network"
+                    },
+                    "x-ms-identifiers": [],
+                    "description": "IPConfigurationReferences - list of IPConfigurationReferences"
+                  },
+                  "routeTable": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
```

Comment:
>id [](http://example.com/codeflow?start=23&length=2)

eTag?

Reply:
So, in order to avoid breaking existing changes, we will want to keep it as "ID" for now. We can change it in the next version of the API.